src/transformStateWithClones.js

```diff
@@ -7,7 +7,26 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = { ...state };
+  const states = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      newState = { ...newState, ...action.extraData };
+    } else if (action.type === 'removeProperties') {
+      const updatedState = { ...newState };
+
+      for (const keyToRemove of action.keysToRemove) {
+        delete updatedState[keyToRemove];
+      }
+      newState = updatedState;
+    } else if (action.type === 'clear') {
+      newState = {};
+    }
```

Comment:
Consider using a switch statement instead of multiple if-else if blocks since you have a limited and known set of conditions to handle. This will make your code cleaner and easier to understand.

Reply:
Consider fixing the AI comment about the switch